src/file_import_retriever.rs

```diff
@@ -0,0 +1,162 @@
+use std::collections::{HashMap, HashSet};
+use std::iter::FromIterator;
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+use swc_common::errors::{ColorConfig, Handler};
+use swc_common::source_map::Pos;
+use swc_ecma_ast::Str;
+use swc_ecma_parser::Capturing;
+use swc_ecma_parser::{lexer::Lexer, Parser, StringInput, Syntax};
+pub struct SourceSpecifiers {
+    specifiers: Vec<swc_ecma_ast::ImportSpecifier>,
+    source: Str,
+}
+
+pub fn get_imports_from_file<'a>(file_path: &'a PathBuf) -> Vec<SourceSpecifiers> {
+    let cm = Arc::<swc_common::SourceMap>::default();
+    let fm = cm
+        .load_file(Path::new(file_path.to_str().unwrap()))
+        .expect("Could not load file");
+    let handler = Handler::with_tty_emitter(ColorConfig::Auto, true, false, Some(cm.clone()));
+
+    let lexer = create_lexer(&fm);
+
+    let capturing = Capturing::new(lexer);
+
+    let mut parser = Parser::new_from(capturing);
+
+    for e in parser.take_errors() {
+        e.into_diagnostic(&handler).emit();
+    }
+
+    let ts_module = parser
+        .parse_typescript_module()
+        .map_err(|e| e.into_diagnostic(&handler).emit())
+        .expect("Failed to parse module.");
```

Comment:
You should handle this error and return an error type here. You could do it in this change or a follow-up PR, but please do it eventually 🙏🥺